build.ps1

```diff
@@ -70,7 +71,8 @@ Param(
     [string] $framework = "*",
     [switch] $skipIotHubTests,
     [switch] $skipDPSTests,
-    [switch] $noBuildBeforeTesting
+    [switch] $noBuildBeforeTesting,
+    [switch] $runSamples
```

Comment:
add a new function to the build script that builds and runs all samples as part of nightly runs